photutils/psf/groupstars.py

```diff
@@ -0,0 +1,228 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""Module which provides classes to perform source grouping."""
+
+from __future__ import division
+import abc
+from astropy.extern import six
+import numpy as np
+from astropy.table import Column, Table, vstack
+
+
+__all__ = ['DAOGroup', 'DBSCANGroup', 'GroupStarsBase']
+
+
+@six.add_metaclass(abc.ABCMeta)
+class GroupStarsBase(object):
```

Comment:
This class and the `group_stars` method needs to have a docstring that lays out explicitly what the interface is (i.e., what must `starlist` be, and what must the method return).  That way a user can implement their own grouping method if they don't like one of the options here.